src/services/services.ts

```diff
@@ -3097,7 +3097,9 @@ namespace ts {
                  oldSettings.noResolve !== newSettings.noResolve ||
                  oldSettings.jsx !== newSettings.jsx ||
                  oldSettings.allowJs !== newSettings.allowJs ||
-                 oldSettings.disableSizeLimit !== oldSettings.disableSizeLimit);
+                 oldSettings.disableSizeLimit !== oldSettings.disableSizeLimit ||

```

Comment:
can you rename `changesInCompilationSettingsAffectSyntax` to `shouldCreateNewSourceFiles`?